Assets/BossRoom/Scripts/Server/Game/Character/ServerCharacterMovement.cs

```diff
@@ -47,20 +47,21 @@ public class ServerCharacterMovement : NetworkBehaviour
         private void Awake()
         {
             m_NavigationSystem = GameObject.FindGameObjectWithTag(NavigationSystem.NavigationSystemTag).GetComponent<NavigationSystem>();
+            // disable this NetworkBehavior until it is spawned
```

Comment:
> might be worth it to summarize your findings in a comment here, else users reading this code might wonder why we do this. 

Good idea, I will. Should I also apply this fix to our other server-only NetworkBehaviours as part of this same PR?

> Will this be added to your sequence diagrams as well?

Yes

Reply:
no
and not in general either, unless that's actually causing issues. Let's not add complexity for nothing.

Reply:
Should this be the proper way of making sure that only the server is running a specific NetworkBehavior, or is it a temporary fix?
If it is the proper way, I feel we should use it on all NetworkBehaviors that are expected to only run on the server, even if they haven't caused exceptions during playtests yet, because some unseen issue could eventually happen and create unwanted behavior that we won't notice. For example, as far as I know, the issue this PR fixes only happened once during a playtest and I have not been able to reproduce it.
On the other hand, if it is a temporary fix, should we even do it in the first place? Wouldn't this cause confusion to our users as to why it was needed here but not in other places?

Reply:
yeah, I agree it's not very clean.
So just to validate my understanding, this is what's happening right now, correct?
0 - object is instantiated during late joining
1 - Awake
2 - Start
3 - FixedUpdate a few times while waiting for additive scene loading to do its thing with other scenes.
4 - OnNetworkSpawn

Normally, without additive scene loading, OnNetworkSpawn would be called right after Start (or between Awake and Start), right?
And so basically, the way we've been doing things is fine as long as you don't have that combination of events happening.
This is basically adding yet another edge case for when OnNetworkSpawn is called.
A solution could be to have OnNetworkSpawn called as soon as the associated additive scene is loaded, but not waiting for all additive scenes to be loaded?
If what I wrote above makes sense, we can poke Noel, see if that'd be too big a change to fix.

Reply:
Yes, that's exactly it. I'll verify this tomorrow, but I think OnNetworkSpawn is normally called after Start. I think doing what you suggest would make it much simplet for users. Although, there might be a reason we need to wait until all scenes are loaded before spawning the in-scene NetworkObjects. I agree we should poke Noel about this

Reply:
@NoelStephensUnity I invoke thee!

Reply:
The issue is really an architectural one.  Under the scenario where there might be a dependency between NetworkObjects, especially if the NetworkObjects exist in different scenes, we need to load all scenes first. One example would be a NetworkPrefabHandler pool in an additively loaded scene is spawning things that get moved into a different scene.  The default when instantiating new GameObjects is they are placed in the currently active, which is typically the single mode loaded scene, but perhaps a user wants a specific additively loaded scene to hold all spawned NetworkObjects of a specific type and automatically migrates them into a completely different additively loaded scene...under these scenario we have to wait for all scenes to be loaded.  Right now this is specific to NetworkPrefabHandler based pools so we could add some kind of flag (or the like) which denotes a dependency (this was my original design), but after some discussion it turned out that the over-all consensus was this would be even more confusing than just loading all scenes first and automatically re-organizing NetworkObjects that are registered in NetworkPrefabHandlers such that they are spawned after the NetworkPrefabHandler itself (this is to make sure the spawning on the client side pulls from the pool and doesn't just instantiate new NetworkObjects prior to the NetworkPrefabHandler being spawned on the client).
_(let me look at this fix and I might be able to provide some suggestions)_

Reply:
So, if I understand this correctly...you basically want only the server to update this component and the client should ignore it (i.e. it gets disabled on the client side)?

My first thought was to just do this:
```
        public override void OnNetworkSpawn()
        {
            if (!IsServer)
            {
                enabled = false;
            }

            // On the server enable navMeshAgent and initialize
            m_NavMeshAgent.enabled = true;
            m_NavPath = new DynamicNavPath(m_NavMeshAgent, m_NavigationSystem);            
        }
        private void FixedUpdate()
        {
            if (IsSpawned)
            {
                PerformMovement();

                m_NetworkCharacterState.MovementStatus.Value = GetMovementStatus();
            }
        }
```
But there is a slight performance hit using the IsSpawned property within the FixedUpdate.  

So, a less expensive version of the above would be:
```
        private bool m_LocalIsSpawned;
        public override void OnNetworkSpawn()
        {
            if (!IsServer)
            {
                enabled = false;
                return;
            }
            m_LocalIsSpawned = true;
            // On the server enable navMeshAgent and initialize
            m_NavMeshAgent.enabled = true;
            m_NavPath = new DynamicNavPath(m_NavMeshAgent, m_NavigationSystem);            
        }

        private void FixedUpdate()
        {
            if (m_LocalIsSpawned)
            {
                PerformMovement();

                m_NetworkCharacterState.MovementStatus.Value = GetMovementStatus();
            }
        }
```
This basically assures that nothing calls PerformMovement until the NetworkObject is considered spawned.

Does this help at all?

Reply:
@LPLafontaineB what do you think?

Reply:
For objects that are placed in-scene would we still have those restrictions regarding NetworkPrefabHandlers? Could it be possible to have them spawn as the scene containing them is loaded?

The solution you suggest works (and I think we do it the same way with another NetworkBehavior to solve a similar issue), but if we are to use a fix, I think it is simpler to just disable the NetworkBehavior in its Awake and enable it in OnNetworkSpawn.

Reply:
Well, there are many ways to accomplish the same thing in Unity which makes it super-cool to use and a challenge for developing the netcode sdk.  With that said, I am planning on doing work in v1.1 that will solve for a few issues in scene synchronization and I could look into adding NetworkObject dependencies back into the mix as it was one of the features I originally had in place to provide more flexibility (at the time I think it was decided to hold off on this feature so it got removed).  The basic idea behind that implementation was that unless a NetworkObject had one or more registered dependencies it would get instantiated and spawned in its GameObject's current scene.  Of course, if a NetworkObject had a dependency then it would wait until all of the associated dependencies' scenes were loaded before it loaded.

So, I will need to get with Louis to discuss adding this to the "menu" for v1.1...as that would fall under the "new feature" classification category.

For now, the decision to be made is whether you want to follow the same pattern as the other place(s) in Boss Room use to solve for this =or= implement a different approach (i.e. like the above) and use both as a way to show users how they can accomplish the same thing.  Both are valid ways as they accomplish the same thing... although if we are talking about performance...disabling the component initially and then enabling it only if you are the server seems like the most "performance-centric" solution.

Reply:
I see. I did not think about NetworkObjects having dependencies in other scenes. I think I understand better now, thank you!

In that case I think it would be a cool feature, but I understand now that it might be less confusing to treat all NetworkObjects the same like how it is right now.

In the meantime, what do you think @SamuelBellomo  about using both solutions to show users there are multiple ways to solve issues? Would it help or should we aim for consistency instead?